edge/pkg/devicetwin/dtmanager/dmiserver.go

```diff
@@ -0,0 +1,361 @@
+package dtmanager
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	"github.com/kubeedge/beehive/pkg/core/model"
+	pb "github.com/kubeedge/kubeedge/edge/pkg/apis/dmi/v1"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dmiclient"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dmiserver"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dtcommon"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dtcontext"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dttype"
+	"github.com/kubeedge/kubeedge/edge/pkg/metamanager/dao"
+	"github.com/kubeedge/kubeedge/pkg/apis/devices/v1alpha2"
+)
+
+// ResourceTypeDeviceModel is type for device model distribute to node
+const (
+	ResourceTypeDeviceModel  = "devicemodel"
```

Comment:
Here are some resourceType that are defined the same as cloudcore, we can extract to the public pkg

Reply:
the resourceTypes are also moved to devicecontroller/constants/device.go